include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -35,25 +35,79 @@ class sycl_backend
     using execution_resource_t = resource_type;
     using resource_container_t = std::vector<execution_resource_t>;
 
+    bool has_enable_profiling = false;
   private:
-    class async_waiter
+    class async_waiter_base{
```

Comment:
What about virtual destructor for classes which will implement this interface?

Reply:
Added a virtual destructor.